lib/pure/strutils.nim

```diff
@@ -1333,30 +1333,53 @@ proc findAux(s, sub: string, start: int, a: SkipTable): int =
     inc(j, a[s[j+m]])
   return -1
 
-proc find*(s, sub: string, start: Natural = 0): int {.noSideEffect,
+when not (defined(nimvm) or defined(js) or defined(nimdoc) or defined(nimscript)):
```

Comment:
'nimvm' is not a define, I think you should simply remove this from the condition.